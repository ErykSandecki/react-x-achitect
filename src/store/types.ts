import { Store } from 'redux';

// types
import { TAction } from 'types/redux';
import { TMainState } from 'types/reducers';

export type TStore = Store<TMainState, TAction<any>> & {
  dispatch: unknown;
};
