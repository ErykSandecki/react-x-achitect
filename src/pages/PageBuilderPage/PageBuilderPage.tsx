import { FC, useState } from 'react';

// components
import Toolbar from 'components/PageBuilder/components/Toolbar/Toolbar';
import ViewBox from 'components/PageBuilder/components/ViewBox/ViewBox';

// hooks
import { useWheelEvent } from './hooks/useWheelEvent';

// others
import { BASE_3D, Box } from 'shared';

// types
import { MouseMode } from 'components/PageBuilder/enums';

const PageBuilderPage: FC = () => {
  const [coordinates, setCoordinates] = useState(BASE_3D);
  const [mouseMode, setMouseMode] = useState(MouseMode.default);

  useWheelEvent();

  return (
    <Box>
      <Toolbar mouseMode={mouseMode} setMouseMode={setMouseMode} />
      <ViewBox
        coordinates={coordinates}
        mouseMode={mouseMode}
        setCoordinates={setCoordinates}
      />
    </Box>
  );
};

export default PageBuilderPage;
