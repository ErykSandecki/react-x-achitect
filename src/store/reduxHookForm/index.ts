export * as actions from './actions';
export * as actionsType from './actionsType';
export * as reducer from './reducer';
export * as selectors from './selectors';
export * as types from './types';
