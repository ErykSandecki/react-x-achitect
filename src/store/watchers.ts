import { Saga, SagaMiddleware } from 'redux-saga';

// store
import { watchAppInitializer } from './appInitializer/watch';

const watchers: Array<Saga<any>> = [watchAppInitializer];

const sagaMiddlewareRuns = (sagaMiddleware: SagaMiddleware): void => {
  for (const watcher of watchers) {
    sagaMiddleware.run(watcher);
  }
};

export default sagaMiddlewareRuns;
