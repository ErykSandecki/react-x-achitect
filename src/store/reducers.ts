import { combineReducers, Reducer } from 'redux';

// store
import appInitializer from './appInitializer/reducer';
import pageBuilder from './pageBuilder/reducer';
import reduxHookForm from './reduxHookForm/reducer';

// types
import { TMainState } from 'types/reducers';

const reducers = {
  appInitializer,
  pageBuilder,
  reduxHookForm,
};

const createRootReducer = (): Reducer<TMainState> => combineReducers(reducers);

export default createRootReducer;
