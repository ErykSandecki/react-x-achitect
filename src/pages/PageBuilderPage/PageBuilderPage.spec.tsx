import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import PageBuilderPage from './PageBuilderPage';

// mocks
import 'test/mocks/sagas/allSagas';

// store
import { configureStore } from 'store/store';

describe('PageBuilderPage snapshots', () => {
  it('should render PageBuilderPage', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <PageBuilderPage />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
