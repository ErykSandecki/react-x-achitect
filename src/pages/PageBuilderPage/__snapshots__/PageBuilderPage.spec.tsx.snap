// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageBuilderPage snapshots should render PageBuilderPage 1`] = `
<DocumentFragment>
  <div
    class="css-q0izro"
    data-test-box=""
  >
    <div
      class="css-1hfqfil"
      data-test-box=""
    >
      <div
        class="css-q0izro MouseModes MouseModes--light"
        data-test-box=""
      >
        <div
          class="css-q0izro MouseModes__button MouseModes__button--light"
          data-test-box=""
        >
          <div
            class="css-q0izro Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="menu"
            height="16"
            name="Logo"
            width="16"
          />
        </div>
        <div
          class="css-q0izro MouseModes__button MouseModes__button--light MouseModes__button--active MouseModes__button--active--light"
          data-test-box=""
        >
          <div
            class="css-q0izro Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="default"
            height="16"
            name="MoveTool"
            width="16"
          />
        </div>
        <div
          class="css-q0izro MouseModes__button MouseModes__button--light"
          data-test-box=""
        >
          <div
            class="css-q0izro Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="toolBeltA"
            height="16"
            name="FrameTool"
            width="16"
          />
        </div>
        <div
          class="css-q0izro MouseModes__button MouseModes__button--light"
          data-test-box=""
        >
          <div
            class="css-q0izro Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="move"
            height="16"
            name="MoveHand"
            width="16"
          />
        </div>
        <div
          class="css-q0izro MouseModes__button MouseModes__button--light"
          data-test-box=""
        >
          <div
            class="css-q0izro Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="comment"
            height="16"
            name="Comment"
            width="16"
          />
        </div>
      </div>
    </div>
    <div
      class="css-17c89wm ViewBox ViewBox--light ZoomBox ZoomBox--light ZoomBox__idle ZoomBox__idle--light"
      data-test-box=""
    >
      <div
        class="css-qel697 ZoomBox__zoom-content ZoomBox__zoom-content--light"
        data-test-box=""
        style="transform: translate(0px, 0px) scale(1);"
      />
    </div>
  </div>
</DocumentFragment>
`;
